multisig-wallet/project/contracts/multisig-contract/src/main.sw

```diff
@@ -126,40 +211,12 @@ impl MultiSignatureWallet for Contract {
             storage.total_weight.write(storage.total_weight.read() - (current_weight - user.weight));
         }
 
-        require(storage.threshold.read() <= storage.total_weight.read(), InitError::TotalWeightCannotBeLessThanThreshold);
+        require(threshold <= storage.total_weight.read(), InitError::TotalWeightCannotBeLessThanThreshold);
 
-        // DRY, if they set the same weight then they pay the extra `write` operation
         storage.weighting.insert(user.address, user.weight);
-        storage.nonce.write(storage.nonce.read() + 1);
+        storage.nonce.write(nonce + 1);
 
-        log(SetWeightEvent { user })
-    }
-
-    #[storage(read, write)]
-    fn transfer(
-        asset_id: ContractId,
-        data: b256,
-        signatures: Vec<SignatureInfo>,
-        to: Identity,
-        value: u64,
-    ) {
-        require(storage.nonce.read() != 0, InitError::NotInitialized);
-        require(value <= this_balance(asset_id), ExecutionError::InsufficientAssetAmount);
-
-        let transaction_hash = hash_transaction(data, storage.nonce.read(), to, value);
-        let approval_count = count_approvals(signatures, transaction_hash);
-        require(storage.threshold.read() <= approval_count, ExecutionError::InsufficientApprovals);
-
-        storage.nonce.write(storage.nonce.read() + 1);
-
-        transfer(value, asset_id, to);
-
-        log(TransferEvent {
-            asset: asset_id,
-            nonce: storage.nonce.read() - 1,
-            to,
-            value,
-        });
+        log(SetWeightEvent { nonce, user })
```

Comment:
As above:
I've included as it's necessary to know the nonce of the wallet in order to execute transactions, so a user can use this logged response instead of having to call the nonce getter when making multiple transactions in sequence.